### PR TITLE
CI nits and updates

### PR DESCRIPTION
**Pull request by kolyshkin**
* Instead of compiling criu from master branch, use a binary package.
* Bump actions/checkout to v3.
* Remove unsupported Go versions, add 1.18.x.
* Bump golangci-lint to v1.45

See individual commits for details.

### CODE REVIEW

**Comment by kolyshkin** [comment]
Note that the required set of jobs has changed so it needs to be updated. I can do that once this is LGTMed right before the merge.

@adrianreber @rst0git PTAL

**Comment by adrianreber** [comment]
Looks good. Although, if you look at the log file, it seems we could skip installing CRIU:
```
criu is already the newest version (3.16.1-3).
criu set to manually installed.
0 upgraded, 0 newly installed, 0 to remove and 145 not upgraded.
```

As the GitHub Actions images pull in Podman, they automatically also pull in CRIU.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -8,15 +8,26 @@ jobs:
...
 
     - name: install criu
+      if: matrix.criu_branch == ''
+      env:
+        REPO: https://download.opensuse.org/repositories/devel:/tools:/criu/xUbuntu_20.04
+      run: |
+        curl -fSsl $REPO/Release.key | sudo apt-key add -
+        echo "deb $REPO/ /" | sudo tee /etc/apt/sources.list.d/criu.list
+        sudo apt update
+        sudo apt install criu
```
Comment (rst0git) [inline .github/workflows/main.yml:27]: @kolyshkin criu is installed by default in the Ubuntu 20.04 GitHub Action virtual environment as dependency Podman.
https://github.com/actions/virtual-environments/blob/21b8cadc/images/linux/scripts/installers/containers.sh

File: .github/workflows/verify.yml [hunk 2]
```diff
@@ -5,18 +5,20 @@ jobs:
   lint:
     runs-on: ubuntu-20.04
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
+    - uses: actions/setup-go@v2
```
Comment (snprajwal) [inline .github/workflows/verify.yml:9]: We can bump this up to v3

File: .github/workflows/verify.yml [hunk 3]
```diff
@@ -5,18 +5,20 @@ jobs:
...
+      with:
+        go-version: 1.18.x
     - uses: golangci/golangci-lint-action@v2
       with:
-        # must be specified without patch version
-        version: v1.42
+        version: v1.45
```
Comment (snprajwal) [inline .github/workflows/verify.yml:14]: We can bump this too up to golangci-lint-action@v3 and use v1.46 with it

**Comment by adrianreber** [comment]
@rst0git @snprajwal Maybe just merge this and do another PR on top of if for additional fixes.

**Comment by rst0git** [comment]
@adrianreber GitHub requires at least two approvals to be able to merge this PR.

**Comment by rst0git** [comment]
I've rebased the patches in this pull request in https://github.com/checkpoint-restore/go-criu/pull/74

**Pull request closed without merging**